metricflow/test/test_utils.py

```diff
@@ -63,3 +65,52 @@ def find_metric_with(model: UserConfiguredModel, function: Callable[[Metric], bo
             return metric, index
 
     raise Exception("Unable to find a metric matching function criteria")
+
+
+def model_with_materialization(
+    initial_model: UserConfiguredModel, materializations: List[Materialization]
+) -> UserConfiguredModel:
+    """Returns a copy of the model but with the materialization added"""
+    model_copy = copy.deepcopy(initial_model)
+    # model_copy.materializations = []
+    model_copy.materializations.extend(materializations)
```

Comment:
I don't like this mechanism. It makes it hard to reason about what's going on when something in the test is broken, because we have a model but then we mutate the instance to be a different model but the same model but it's also a different copy.

That said, we should not be trying to fix this here.

Reply:
A lot of our model fixtures are scoped to the test session meaning that the model fixtures are persisted over the entirety of the test run (i.e. it doesn't get destroyed and recreated between tests). So we copy the model so that it doesn't affect the fixture for other tests. Another solution would be to switch the scope of the fixture to the default of `function` which means it would get destroyed and recreated between different test functions. I think we do it the current way to save time? I'm not wholly sure. 

Anyway, the model that is passed in is a model pre-transformations (i.e. it exists in the base state of a model which has been read from yaml files but hasn't been validated and transformed yet). So adding to the model is basically saying we need this for the state of this test, but not all other tests.